applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1560,21 +1585,22 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateAndAddFluidBodyFlow(Vector
 }
 
 template <unsigned int TDim, unsigned int TNumNodes>
-void UPwSmallStrainElement<TDim, TNumNodes>::CalculateStrain(ElementVariables& rVariables, unsigned int GPoint)
+Vector UPwSmallStrainElement<TDim, TNumNodes>::CalculateStrain(const Matrix& rDeformationGradient,
+                                                               const Matrix& rB,
+                                                               const Vector& rDisplacements,
+                                                               bool          UseHenckyStrain)
 {
-    if (rVariables.UseHenckyStrain) {
-        rVariables.F    = this->CalculateDeformationGradient(GPoint);
-        rVariables.detF = MathUtils<>::Det(rVariables.F);
-        noalias(rVariables.StrainVector) = StressStrainUtilities::CalculateHenckyStrain(rVariables.F, VoigtSize);
+    if (UseHenckyStrain) {
+        return StressStrainUtilities::CalculateHenckyStrain(rDeformationGradient, VoigtSize);
     } else {
-        this->CalculateCauchyStrain(rVariables);
+        return this->CalculateCauchyStrain(rB, rDisplacements);
     }
 }
```

Comment:
We could simplify this even further by either using the ternary operator or by removing the explicit `else` keyword:
```
return UseHenckyStrain ? StressStrainUtilities::CalculateHenckyStrain(rDeformationGradient, VoigtSize)
                       : this->CalculateCauchyStrain(rB, rDisplacements);
```
Or, if that is too compact, use this:
```
if (UseHenckyStrain) {
    return StressStrainUtilities::CalculateHenckyStrain(rDeformationGradient, VoigtSize);
}

return this->CalculateCauchyStrain(rB, rDisplacements);
```

Reply:
Agreed, makes it cleaner!